docs/source/format/ADBC.rst

```diff
@@ -0,0 +1,310 @@
+Arrow, for C, Go, and Java (with bindings for other languages).
+Applications code to this API standard (in much the same way as they
+would with JDBC or ODBC), but fetch result sets in Arrow format
+(e.g. via the :doc:`C Data Interface <./CDataInterface>`).  They then
+link to an implementation of the standard: either directly to a
+vendor-supplied driver for a particular database, or to a driver
+manager that abstracts across multiple drivers.  Drivers implement the
+standard using a database-specific API, such as Flight SQL.
+
+Goals
+-----
+
+- Provide a cross-language, Arrow-based API to standardize how clients
+  submit queries to and fetch Arrow data from databases.
+- Support both SQL dialects and the emergent `Substrait`_ standard.
+- Support explicitly partitioned/distributed result sets to work
+  better with contemporary distributed systems.
+- Allow for a variety of implementations to maximize reach.
+
+Non-goals
+---------
+
+- Replacing JDBC/ODBC in all use cases, particularly `OLTP`_ use
+  cases.
+- Requiring or enshrining a particular database protocol for the Arrow
+  ecosystem.
+
+ADBC API Standard 1.0.0
+=======================
+
+ADBC is a C header that can be implemented directly by a
+vendor-specific "driver" or a vendor-neutral "driver manager".  The
+header is reproduced in full at the end of this document, and is
+intended to be self-documenting.
+
+Example use cases
+-----------------
+
+A C or C++ application wishes to retrieve bulk data from a Postgres
+database for further analysis.  The application is compiled against
+the ADBC header, and executes queries via the ADBC APIs.  The
+application is linked against the ADBC libpq driver.  At runtime, the
+driver submits queries to the database via the Postgres client
+libraries, and retrieves row-oriented results, which it then converts
+to Arrow format before returning them to the application.
+
+If the application wishes to retrieve data from a database supporting
+Flight SQL instead, it would link against the ADBC Flight SQL driver.
+At runtime, the driver would submit queries via Flight SQL and get
+back Arrow data, which is then passed unchanged and uncopied to the
+application.  (The application may have to edit the SQL queries, as
+ADBC does not translate between SQL dialects.)
+
+If the application wishes to work with multiple databases, it would
+link against the ADBC driver manager, and specify the desired driver
+at runtime.  The driver manager would pass on API calls to the correct
+driver, which handles the request.
+
+In Go and Java
+--------------
+
+In these languages, ADBC consists of a set of interfaces in a
+canonical package.
+
+For Go, version 1.0.0 of the standard corresponds to release 1.0.0 of
+the package ``github.com/apache/arrow-adbc/go/adbc``.
+
+.. TODO: this is not yet tagged
+
+For Java, version 1.0.0 of the standard corresponds to release 1.0.0
+of the package ``org.apache.arrow.adbc:adbc-core``.
+
+.. TODO: this is not yet tagged
```

Comment:
Should this also reference / mention the initial implementations that are also being released in those libraries? Such as the driver manager / the Go wrapper for ADBC interfaces to work with the standard `database/sql` package etc.?

Reply:
I mean for this to be only about the core definitions, not the packages (which are going to update more frequently)

Reply:
We should figure out where and how to document ADBC itself, though

Reply:
The Go package will end up on pkg.go.dev but we should probably add something with the rest of the arrow docs so it's findable